core/chain_makers.go

```diff
@@ -403,4 +431,9 @@ func (cr *fakeChainReader) GetHeaderByNumber(number uint64) *types.Header
 func (cr *fakeChainReader) GetHeaderByHash(hash common.Hash) *types.Header          { return nil }
 func (cr *fakeChainReader) GetHeader(hash common.Hash, number uint64) *types.Header { return nil }
 func (cr *fakeChainReader) GetBlock(hash common.Hash, number uint64) *types.Block   { return nil }
-func (cr *fakeChainReader) GetTd(hash common.Hash, number uint64) *big.Int          { return nil }
+func (cr *fakeChainReader) GetTd(hash common.Hash, number uint64) *big.Int {
+	if cr.config.TerminalTotalDifficultyPassed {
+		return cr.config.TerminalTotalDifficulty
```

Comment:
This seems wrong given that the terminal difficulty will be larger most of the times. But returning `nil` before the merge seems weird too. Do we even use this anywhere?

Reply:
The problem is that with `fakeChainReader` always returning `nil`, the beacon consensus engine will always fall back to the ethash difficulty computation. I added a fake beacon engine to simulate merged networks in tests